### PR TITLE
NFDIV-1909 - Update IrretrievablyBroken question API mapping

### PR DESCRIPTION
**Pull request by amoriartyMOJ**
### Change description ###

As the field for marriageIrretrievablyBroken has changed on the back end
from a YesOrNo field to a Set, this PR will update the mapping of the field.

Update definition.ts which was generated by the new API version.

Add new mapping to and from API for applicant1ScreenHasUnionBroken.

Fix issues found due to update in definition.ts file.

Update unit tests.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-1909

**Before merging a pull request make sure that:**

- [X] tests have been updated / new tests has been added (if needed)
- [X] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

### CODE REVIEW

File: src/test/steps/common.ts [hunk 1]
```diff
@@ -195,10 +194,6 @@ When('I enter my valid case reference and valid access code', async () => {
   iClick('Continue');
 });
 
-When('a case worker issues the application', async () => {
-  await triggerAnEvent(CASEWORKER_ISSUE_APPLICATION, { ceremonyPlace: 'Somewhere' });
-});
-
```
Comment (lara-mpt) [inline src/test/steps/common.ts:201]: Why remove this? It's used in a couple of places still

File: src/main/steps/applicant1/irretrievable-breakdown/post.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    formData.divorceOrDissolution = req.session.userCase.divorceOrDissolution;
+
+    if (req.body.saveAndSignOut || req.body.saveBeforeSessionTimeout) {
+      await this.saveAndSignOut(req, res, formData);
+    } else {
+      await this.saveAndContinue(req, res, form, formData);
+    }
+  }
+}
```
Comment (lara-mpt) [inline src/main/steps/applicant1/irretrievable-breakdown/post.ts:24]: I'm not sure this needs a custom post controller, the only difference is this sets the divorceOrDissolution field in the formData, but I can't see why we'd need to do that?

Comment (amoriartyMOJ) [reply src/main/steps/applicant1/irretrievable-breakdown/post.ts:24]: discussed and agreed this is the right approach as without this the divorceOrDissolution field is undefined and this causes the conversion to API to fail as it's required in order to convert correctly. Marking as resolved.

File: src/test/steps/common.ts (hunk 1 above)
Comment (amoriartyMOJ) [reply src/test/steps/common.ts]: After chatting yesterday, this has been manually added back to get the PR working. We will be investigating why the event was removed in the first place but this PR can still progress.

**Review by lara-mpt — APPROVED** [review]

File: src/main/app/case/from-api-format.ts [hunk 3]
```diff
@@ -19,6 +28,32 @@ const checkboxConverter = (value: string | undefined) => {
   return value === YesOrNo.YES ? Checkbox.Checked : Checkbox.Unchecked;
 };
 
+const marriageBrokenConverter = data => {
```
Comment (benouaer) [inline src/main/app/case/from-api-format.ts:31]: do we need two converters here? could we not combine them like you do in the to-api, given that it's either DivorceOrDissolution.DIVORCE or DivorceOrDissolution.DISSOLUTION

Comment (amoriartyMOJ) [reply src/main/app/case/from-api-format.ts:31]: From my understanding as there are 2 API fields, each one will call the same conversion which will end up with the same function being called twice if it's combined, which could change the value depending on the order it's called?

Comment (benouaer) [reply src/main/app/case/from-api-format.ts:31]: I think the values would still match, no?

**Comment by amoriartyMOJ** [comment]
Closing until requested.

**Pull request closed without merging**